### PR TITLE
Handling SslException on channel close

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
This issue was reported by a customer (ICM#205604603) - adding changes to handle this error and reduce noise level in logs.

Once we have this issue resolved, we would need you to take a look at the below exception. 

Customer is getting hundreds of these exceptions while performing write operations. They are only seeing these errors in the logs and there is no impact to the application.

javax.net.ssl.SSLException: SSLEngine closed already
                at io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:854)
                at io.netty.handler.ssl.SslHandler.wrapAndFlush(SslHandler.java:811)
                at io.netty.handler.ssl.SslHandler.flush(SslHandler.java:792)
                at io.netty.handler.ssl.SslHandler.flush(SslHandler.java:1939)
                at io.netty.handler.ssl.SslHandler.closeOutbound0(SslHandler.java:668)
                at io.netty.handler.ssl.SslHandler.closeOutbound(SslHandler.java:652)
                at io.netty.handler.ssl.SslHandler.closeOutbound(SslHandler.java:640)
                at com.azure.cosmos.implementation.directconnectivity.rntbd.RntbdRequestManager.close(RntbdRequestManager.java:391)
                at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622)
                at io.netty.channel.AbstractChannelHandlerContext.access$1200(AbstractChannelHandlerContext.java:61)
                at io.netty.channel.AbstractChannelHandlerContext$11.run(AbstractChannelHandlerContext.java:611)
                at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
                at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
                at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
                at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
                at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
                at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
                 at java.lang.Thread.run(Thread.java:748)

### CODE REVIEW

**Review by kirankumarkolli — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java [hunk 1]
```diff
@@ -391,10 +392,30 @@ public void close(final ChannelHandlerContext context, final ChannelPromise prom
...
+                // This ensures that all SSL engine and ByteBuf resources are released
+                // This is something that does not occur in the call to ChannelPipeline.close that follows
+                sslHandler.closeOutbound();
+            } catch (Exception exception) {
+
+                // Netty will throw the following exception here if the outbound SSL connection has been closed already
+                // javax.net.ssl.SSLException: SSLEngine closed already
+                // Reducing the noise level here because multiple concurrent closes can happen due to race conditions
+                // and there is no harm in this case
+                if (exception instanceof SSLException) {
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:407]: Do similar happen for reactor http code paths also?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:407]: For ReactorNettyClient Kushagra had some discussions with the owner of ReatorNettyClient - https://github.com/reactor/reactor-netty/issues/1165 

My understanding reading through the work item is that this addressed the issues - but I will double-check with @kushagraThapar  when he is back next week

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java [hunk 2]
```diff
@@ -391,10 +392,30 @@ public void close(final ChannelHandlerContext context, final ChannelPromise prom
...
+                    logger.debug(
+                        "SslException when attempting to close the outbound SSL connection: {}",
+                        exception.toString());
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:410]: you don't need "{}" for exception and also exception.toString not needed here.

```suggestion
logger.debug("SslException when attempting to close the outbound SSL connection", exception);
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java [hunk 3]
```diff
@@ -391,10 +392,30 @@ public void close(final ChannelHandlerContext context, final ChannelPromise prom
...
+                } else {
+                    logger.warn(
+                        "Exception when attempting to close the outbound SSL connection: {}",
+                        exception.toString());
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:414]: ditto.

**Review by moderakh — APPROVED** [review]
one minor comment. LGTM.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:410]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:414]: Fixed

**Pull request merged**